sysvinit/freebsd/dsidentify.tmpl

```diff
@@ -17,10 +17,19 @@ PATH="{{ prefix }}/sbin:{{ prefix }}/bin:/usr/sbin:/usr/bin:/sbin:/bin"
 
 name="dsidentify"
 command="{{ prefix }}/lib/cloud-init/ds-identify"
+start_precmd="dsidentify_precmd"
 start_cmd="dsidentify_start"
 stop_cmd=":"
 rcvar="cloudinit_enable"
 
+dsidentify_precmd()
+{
+    if ! test -L /run ; then
+       rm -rf /run
+       ln -s /var/run /run
```

Comment:
The symlink won't ever be created in the case that /run does not yet exist, since it happens inside of this conditional. Isn't this conditional a workaround for images that have already ran `ds-identify` which created this directory?

In other words, don't we want `/run` to be created as a symlink to `/var/run/` on first boot when no `/run/` exists yet?

I'm thinking that we want something more akin to:
```sh
link_run_to_ephemeral()
{
    if test -L /run; then
       return 0
    elif test -e /run; then
       rm -rf /run
    fi 
    ln -s /var/run /run
}
```

question applies to elsewhere too

Reply:
from http://man.openbsd.org/test.1#L (using OpenBSD because it's easiest to link exactly where i want to, they all say the same)

> -L file
>    True if file exists and is a symbolic link. This operator is for compatibility purposes. Do not rely on its existence; use -h instead.